pkg/scheduler/scheduler.go

```diff
@@ -368,6 +370,36 @@ var defaultQueueingHintFn = func(_ klog.Logger, _ *v1.Pod, _, _ interface{}) (fr
 	return framework.Queue, nil
 }
 
+// TODO: remove this function once all plugins implement the QueueingHintFn for podUpdate event
+// and SchedulerQueueingHints feature gate hits GA
+// defaultQueueingHintFnPodUpdate is the default QueueingHintFn for pod related event.
+// It returns queue on podAdd/Delete events. And do a simple check for podUpdate event.
+var defaultQueueingHintFnPodUpdate = func(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
Why do we need this?

Reply:
1. In this pr, we'll remove old function `isPodUpdated` , and when an unschedulable pod is updated, we'll put the pod into the corresponding queue by it's unschedulable plugins' queueHint function.
2. When feature gate is disabled, we'll ignore the functions implemented by plugins and just register defaultFn for each plugin, in L423.

So before, we'll also register defaultQueueingHintFn for podUpdate events.

- PodA is rejected by pluginA
- Scheduler update podA's conditions
- PodUpdate events is triggered, schedulerQueue judge pod by pluginA's queueHintFns
- defaultQueueingHintFn returns queue, thus podA is put into the backOffQueue
- Endless loop

I think podUpdate is a unique event, and should be took care of here.

Reply:
Not all true. We'll also use it when ft is enabled, but a plugin registered podUpdate event without queueHint.

Reply:
🙆‍♂️  I'm convinced.